lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1005,7 +1005,7 @@ bool TypeChecker::typeCheckCatchPattern(CatchStmt *S, DeclContext *DC) {
 
 static bool isDiscardableType(Type type) {
   return (type->is<ErrorType>() ||
-          type->isNever() ||
+          type->isUninhabited() ||
```

Comment:
Yes, sorry, this isn't about whether `Optional<Never>` is inhabited; it's that we shouldn't warn about an unused result here.